src/Microsoft.VisualStudio.Threading.Tests/TplExtensionsTests.cs

```diff
@@ -151,12 +151,47 @@ public void ApplyResultToPreCompletedNonGeneric()
         [Fact]
         public void WaitWithoutInlining()
         {
+            var sluggishScheduler = new SluggishInliningTaskScheduler();
             var originalThread = Thread.CurrentThread;
-            var task = Task.Run(delegate
-            {
-                Assert.NotSame(originalThread, Thread.CurrentThread);
-            });
+            var task = Task.Factory.StartNew(
+                delegate
+                {
+                    Assert.NotSame(originalThread, Thread.CurrentThread);
+                },
+                CancellationToken.None,
+                TaskCreationOptions.None,
+                sluggishScheduler);
+
+            // Schedule the task such that we'll be very likely to call WaitWithoutInlining
+            // *before* the task is scheduled to run on its own.
+            sluggishScheduler.ScheduleTasksLater();
+
+            task.WaitWithoutInlining();
+        }
+
+        [Fact]
+        public void WaitWithoutInlining_DoesNotWaitForOtherInlinedContinuations()
+        {
+            var sluggishScheduler = new SluggishInliningTaskScheduler();
+
+            var task = Task.Factory.StartNew(
+                delegate
+                {
+                },
+                CancellationToken.None,
+                TaskCreationOptions.None,
+                TaskScheduler.Default);
+            var continuationUnblocked = new ManualResetEventSlim();
+            var continuationTask = task.ContinueWith(
+                delegate
+                {
+                    continuationUnblocked.Wait();
+                },
+                CancellationToken.None,
+                TaskContinuationOptions.ExecuteSynchronously,
+                sluggishScheduler); // ensures the continuation never runs unless inlined
             task.WaitWithoutInlining();
+            continuationUnblocked.Set();
         }
```

Comment:
Ya, that's worth something, particularly since we have stress tests that are noisy when run after other tests. We probably bleed stray tasks from hundreds of tests though. I guess I'll block on it. Not a big deal.